pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java

```diff
@@ -57,7 +57,7 @@ public String getServerVersion() {
     public int getServerVersionNum() {
         if (serverVersionNum != 0)
             return serverVersionNum;
-        return Utils.parseServerVersionStr(serverVersion);
+        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
```

Comment:
@valgog , basically, all the "setter" methods (`setStandardConformingStrings`, `addWarning`, ..) are used from `org.postgresql.core.v2.QueryExecutorImpl#interpretCommandStatus` and similar only -> thus no synchronization is required. So removing `synchronized` seems to be a valid optimization.